client/src/portal/mod.rs

```diff
@@ -609,4 +659,29 @@ mod tests {
 
         Ok(())
     }
+
+    #[tokio::test]
+    async fn change_secret() -> Result<(), Error> {
+        let path = PathBuf::from("../../tests/test_rekeying.keyring");
+
+        let mut keyring = Keyring::load(&path, strong_key()).await?;
+        let attributes = HashMap::from([("attr", "value")]);
+        keyring
+            .create_item("test", &attributes, "password", false)
+            .await?;
+        let item_before = keyring.lookup_item(&attributes).await?;
+
+        let new_secret = Secret::from(b"password".to_vec());
+        keyring.change_secret(new_secret).await?;
+
+        let new_secret = Secret::from(b"password".to_vec());
+        let keyring = Keyring::load(&path, new_secret).await?;
+        let item_now = keyring.lookup_item(&attributes).await?;
+
+        assert_eq!(item_before.unwrap().label(), item_now.unwrap().label());
```

Comment:
The labels are not encrypted so this check alone is not enough. you should also check if the secret is still readable and the same for attributes.

Reply:
yeah! done, thanks!